src/components/map/WmtsService.js

```diff
@@ -0,0 +1,163 @@
+              .GetTile
+              .DCP
+              .HTTP
+              .Get[0]
+              .Constraint[0]
+              .AllowedValues
+              .Value[0];
+
+          var layerOptions = {
+            layer: getCapLayer.Identifier,
+            requestEncoding: requestEncoding
+          };
+          getCapLayer.sourceConfig = ol.source.WMTS.optionsFromCapabilities(
+              getCapabilities, layerOptions);
+          getCapLayer.attribution =
+              getCapabilities.ServiceProvider.ProviderName;
+          getCapLayer['attributionUrl'] =
+              getCapabilities.ServiceProvider.ProviderSite;
+          getCapLayer['capabilitiesUrl'] = getCapabilities.OperationsMetadata
+              .GetCapabilities
+              .DCP
+              .HTTP
+              .Get[0]
+              .href;
+
+          return {
+            attributions: getCapLayer.attributions,
+            attributionUrl: getCapLayer.attributionUrl,
+            capabilitiesUrl: getCapLayer.capabilitiesUrl,
+            label: getCapLayer.Title,
+            layer: getCapLayer.Identifier,
+            timestamps: getTimestamps(getCapLayer),
+            sourceConfig: getCapLayer.sourceConfig,
+          };
+        };
+
+        this.getLayerConfigFromIdentifier = function(getCapabilities,
+            identifier) {
+          var layerConfig;
+          if (getCapabilities.Contents && getCapabilities.Contents.Layer) {
+            getCapabilities.Contents.Layer.forEach(function(layer) {
+              if (layer.Identifier === identifier) {
+                layerConfig = getLayerConfig(getCapabilities, layer);
+              }
+            });
+          }
+
+          return layerConfig;
+        };
+
+        // Create an ol WMTS layer from GetCapabilities informations
+        var createWmtsLayer = function(options) {
+          var source = new ol.source.WMTS(options.sourceConfig);
+          var projection = source.getProjection();
+          var extent;
+          if (projection) {
+            extent = projection.getExtent();
+          } else {
+            extent = gaGlobalOptions.defaultExtent;
+          }
+          var layer = new ol.layer.Tile({
+            id: 'WMTS||' + options.layer + '||' +
+              options.capabilitiesUrl,
+            source: source,
+            extent: gaMapUtils.intersectWithDefaultExtent(extent),
+            preload: gaMapUtils.preload,
+            attribution: options.sourceConfig.attribution
+          });
+          gaDefinePropertiesForLayer(layer);
+          layer.useThirdPartyData =
+              gaUrlUtils.isThirdPartyValid(options.sourceConfig.urls[0]);
+          layer.label = options.label;
+          layer.url = options.attributionUrl;
+          layer.attributions = options.attributions;
+          layer.timestamps = options.timestamps;
+          layer.timeEnabled = layer.timestamps ?
+              layer.timestamps.length > 1 : false;
+          layer.time = options.timestamp;
```

Comment:
Yes of course you're right. But can your just call it `time` instead of `timestamp`.